notebooks/measures/Search_Download_and_Plot_GeoTIFFs.ipynb

```diff
@@ -0,0 +1,503 @@
+   "id": "16184745",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vel_raster = gdal.Open(vel_fp)\n",
+    "geoTransform = vel_raster.GetGeoTransform()\n",
+    "proj=osr.SpatialReference(wkt=vel_raster.GetProjection())\n",
+    "epsg = 'EPSG:' + proj.GetAttrValue('AUTHORITY',1)\n",
+    "\n",
+    "pixelSizeX = geoTransform[1]\n",
+    "pixelSizeY = geoTransform[5]\n",
+    "minx = geoTransform[0]\n",
+    "maxy = geoTransform[3]\n",
+    "maxx = minx + pixelSizeX * vel_raster.RasterXSize\n",
+    "miny = maxy + pixelSizeY * vel_raster.RasterYSize\n",
+    "\n",
+    "kwargs = { 'format': 'GTiff', 'outputBounds': [minx, miny, maxx, maxy], 'outputBoundsSRS': epsg, 'xRes': pixelSizeX, 'yRes': pixelSizeY}\n",
+    "ds = gdal.Warp(dem_crop, dem_fp, **kwargs)\n",
+    "ds=None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c743d341",
+   "metadata": {},
+   "source": [
+    "### **Set up the grids to plot the DEM and velocity data**\n",
+    "To plot the cropped and downsampled DEM with the velocity data, we need to set up a grid. We will read in the data and metadata using the `open` and `meta` methods, then we will use the `transform` or `affine` methods to map the image coordinates to world coordinates, all methods from the `rasterio` package. This information will then be used to generate x and y grid locations using the `arrange` method from the `numpy` package. It will also mask the no data values from the velocity GeoTIFF to produce a nicer plot. The cell below will print out the metadata from both files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "15fc1119",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# read in subsetted and resampled DEM\n",
+    "dem_src = rasterio.open(dem_crop)\n",
+    "\n",
+    "# print out metadata information\n",
+    "for k in dem_src.meta:\n",
+    "    print(k,dem_src.meta[k])\n",
+    "\n",
+    "# Retrieve the affine transformation\n",
+    "if isinstance(dem_src.transform, Affine):\n",
+    "     transform = dem_src.transform\n",
+    "else:\n",
+    "     transform = dem_src.affine\n",
+    "\n",
+    "N = dem_src.width\n",
+    "M = dem_src.height\n",
+    "dx = transform.a\n",
+    "dy = transform.e\n",
+    "minx = transform.c\n",
+    "maxy = transform.f\n",
+    "\n",
+    "# Read the image data, flip upside down if necessary\n",
+    "dem_crop_in = dem_src.read(1)\n",
+    "if dy < 0:\n",
+    "    dy = -dy\n",
+    "    dem_crop_in = np.flip(dem_crop_in, 0)\n",
+    "\n",
+    "#Uncomment the line below if you wish to see the min/max DEM values\n",
+    "#print('Data minimum, maximum = ', np.amin(data_in), np.amax(data_in))\n",
+    "\n",
+    "# Generate X and Y grid locations\n",
+    "xdata = minx + dx/2 + dx*np.arange(N)\n",
+    "ydata = maxy - dy/2 - dy*np.arange(M-1,-1,-1)\n",
+    "\n",
+    "dem_extent = [xdata[0], xdata[-1], ydata[0], ydata[-1]]\n",
+    "\n",
+    "### read in the velocity data\n",
```

Comment:
```suggestion
    "# read in the velocity data\n",
```

Reply:
These in-line comments are great! If you could include the same thing above, that would be very useful for those who are learning these projection commands.

Reply:
Good idea, have made those additions